feat(service): add librechat template

GauthierPLM opened the pull request:
## Changes
- Add template to deploy [LibreChat](https://www.librechat.ai/)
- Add librechat.svg icon

## Issues
- fix #2867

GauthierPLM committed:
feat: add librechat template

GauthierPLM committed:
fix: incorrect RAG_API_URL variable and missing volume

andrasbacsai commented:
It was a mistake, sorry. I recreated the `next` branch and it auto-closed this PR.

peaklabs-dev commented on templates/compose/librechat.yaml:
```diff
@@ -0,0 +1,458 @@
+      - DB_NAME=rag
+      - RAG_PORT=8000
+      - RAG_OPENAI_API_KEY=${SERVICE_OPENAI_API_KEY:-user_provided}
+    healthcheck:
```
This `healthcheck` does not seem to work, please fix it.

GauthierPLM replied:
Fixed by switching to a Python command. Neither wget or curl are installed in the docker image.

peaklabs-dev commented on templates/compose/librechat.yaml:
```diff
@@ -0,0 +1,458 @@
+      - DB_PASSWORD=${SERVICE_PASSWORD_POSTGRES}
+      - DB_NAME=rag
+      - RAG_PORT=8000
+      - RAG_OPENAI_API_KEY=${SERVICE_OPENAI_API_KEY:-user_provided}
```
If this variable is required add it like so `RAG_OPENAI_API_KEY=${SERVICE_OPENAI_API_KEY:?}`

GauthierPLM replied:
`user_provided` is a special variable that, when set, let users set their own API key to use directly from the UI. But thanks for the tip about `?`. :)

peaklabs-dev commented on templates/compose/librechat.yaml:
```diff
@@ -0,0 +1,458 @@
+      timeout: 20s
+      retries: 10
+
+volumes:
```
This section can be removed as Coolify will add this.

peaklabs-dev commented on templates/compose/librechat.yaml:
```diff
@@ -0,0 +1,458 @@
+    image: getmeili/meilisearch:v1.12.3
+    environment:
+      - MEILI_MASTER_KEY=${SERVICE_PASSWORD_MEILI}
+      - MEILI_NO_ANALYTICS=true
```
Make this variable configurable

peaklabs-dev commented on templates/compose/librechat.yaml:
```diff
@@ -0,0 +1,458 @@
+    environment:
+      - MONGO_INITDB_ROOT_USERNAME=${SERVICE_USER_MONGO}
+      - MONGO_INITDB_ROOT_PASSWORD=${SERVICE_PASSWORD_MONGO}
+      - ALLOW_EMPTY_PASSWORD=${MONGO_ALLOW_EMPTY_PASSWORD:-yes}
```
The password should never be empty so remove this ENV

peaklabs-dev commented on templates/compose/librechat.yaml:
```diff
@@ -0,0 +1,458 @@
+      - MONGO_INITDB_ROOT_USERNAME=${SERVICE_USER_MONGO}
+      - MONGO_INITDB_ROOT_PASSWORD=${SERVICE_PASSWORD_MONGO}
+      - ALLOW_EMPTY_PASSWORD=${MONGO_ALLOW_EMPTY_PASSWORD:-yes}
+    image: mongo
```
Please use a specific image tag with a version not just `mongo`

peaklabs-dev commented on templates/compose/librechat.yaml:
```diff
@@ -0,0 +1,458 @@
+      - librechat-images:/app/client/public/images
+      - librechat-logs:/app/api/logs
+      - librechat-uploads:/app/uploads
+      - type: bind
```
Overall this config is a bit bloated and a lot of stuff is comment out, maybe leave a docs like on top of the config as a comment and remove everything else except what is absolutely needed to clean up the compose file.

GauthierPLM replied:
LibreChat' devs advised to embed the whole file, but I agree with you, it makes the compose file much less maintainable. I kept only the required lines.

peaklabs-dev commented on templates/compose/librechat.yaml:
```diff
@@ -0,0 +1,458 @@
+              # externalUrl: 'https://librechat.ai/privacy-policy'
+              # openNewTab: true
+
+            # Terms of service
```
Not sure if this is actually needed so maybe the Terms of service can be removed

peaklabs-dev commented on templates/compose/librechat.yaml:
```diff
@@ -0,0 +1,458 @@
+services:
+  librechat:
+    image: ghcr.io/danny-avila/librechat-dev-api:latest
+    depends_on:
```
Please make sure it is also `healthy` and not just depending on these containers.

GauthierPLM commented:
Hello,

Thanks for the deep review. I’m currently unable to do any change until mid-June. I’ll work on them as soon as possible after June 16.

peaklabs-dev committed:
Merge branch 'next' into librechat

GauthierPLM committed:
feat(librechat): improve template from review' feedback

- Updated `depends_on` to include `condition: service_healthy` for `mongodb`, `rag_api`, and `vectordb` services.
- Reduced extensive `librechat.yaml` default content to reduce template size.
- Updated `mongodb` image to `mongo:8`.
- Ensured dynamic default values for `MEILI_NO_ANALYTICS` using an environment variable.
- Fixed `rag_api` health check by switching to a Python-based solutio and adjusted health check timeout for `rag_api`.
- Removed `volumes` section.

GauthierPLM commented:
@peaklabs-dev I edited the template according to your review. Hope it will work now. :)

peaklabs-dev committed:
Merge branch 'next' into librechat

DanOpcode commented:
Is there anything left? I'm pumped up to see it merged 🍹

siscia commented:
@peaklabs-dev - I would love to get this merged as well.

I am available to fix whatever is left to fix - but it is unclear if there is anything left or just waiting for capacity to get it merged.

Please feel free to ping me as well if there is any need of a change.

And thanks to the amazing work on coolify.

@GauthierPLM - thanks for the PR! Very much appreciated.

peaklabs-dev reviewed (approved):
Thanks for the PR!

I will clean it up and format it a bit once it is merged.

theabhinavdas commented:
@GauthierPLM thanks a ton for this contribution! QQ—I still don't see this service in my list of options on my coolify instance.. and there seem to be no updates available! How do I install this? cc @peaklabs-dev

peaklabs-dev commented:
@theabhinavdas This service will appear in the services once this PR https://github.com/coollabsio/coolify/pull/6171 has been released.
